macros/utils/table_operations/insert_rows.sql

```diff
@@ -0,0 +1,79 @@
+{% macro insert_rows(table_relation, rows, should_commit=false) %}
+    {% if not table_relation %}
+        {{ elementary.edr_log('Recieved table relation is not valid (make sure elementary models were executed successfully first)') }}
+        {{ return(none) }}
+    {% endif %}
+
+    {% set columns = adapter.get_columns_in_relation(table_relation) -%}
+    {% if not columns %}
+        {% set table_name = elementary.relation_to_full_name(table_relation) %}
+        {{ elementary.edr_log('Could not extract columns for table - ' ~ table_name ~ ' (might be a permissions issue)') }}
+        {{ return(none) }}
+    {% endif %}
+
+    {% set insert_rows_queries = elementary.get_insert_rows_queries(table_relation, columns, rows) %}
+    {% for insert_query in insert_rows_queries %}
+      {% do elementary.debug_log("[%d/%d] Running insert query." % (loop.index, insert_rows_queries | length)) %}
+      {% do dbt.run_query(insert_query) %}
+    {% endfor %}
+    {%- if should_commit -%}
+        {% do adapter.commit() %}
+    {%- endif -%}
+{% endmacro %}
+
+{% macro get_insert_rows_queries(table_relation, columns, rows, query_max_size=none) -%}
+    {% if not query_max_size %}
+      {% set query_max_size = elementary.get_config_var('query_max_size') %}
+    {% endif %}
+
+    {% set insert_queries = [] %}
+    {% set insert_query %}
+       insert into {{ table_relation }}
+         ({%- for column in columns -%}
+           {{- column.name -}} {{- "," if not loop.last else "" -}}
+         {%- endfor -%}) values
+    {% endset %}
+
+    {% set current_query = namespace(data=insert_query) %}
+    {% for row in rows %}
+      {% set rendered_column_values = [] %}
+      {% for column in columns %}
+        {% set column_value = elementary.insensitive_get_dict_value(row, column.name) %}
+        {% do rendered_column_values.append(elementary.render_value(column_value)) %}
+      {% endfor %}
+      {% set row_sql = "(%s)" % (rendered_column_values | join(",")) %}
+      {% set query_with_row = current_query.data + ("," if not loop.first else "") + row_sql %}
+
+      {% if query_with_row | length > query_max_size %}
```

Comment:
A small edge case here - if the query_max_size is smaller than the insert_query + first row - insert_queries will contain an invalid query with only the boiler plate. This is an edge case but could maybe happen if you have a very long first row so might be better to fix it I think.